pkg/encryption/elgamal/encryption.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package elgamal
+
+import (
+	"crypto/ecdsa"
+	"hash"
+
+	"github.com/ethersphere/bee/pkg/crypto"
+	"github.com/ethersphere/bee/pkg/encryption"
+)
+
+func New(key *ecdsa.PrivateKey, pub *ecdsa.PublicKey, salt []byte, padding int, hashfunc func() hash.Hash) (encryption.Interface, error) {
+	dh := crypto.NewDH(key)
+	sk, err := dh.SharedKey(pub, salt)
+	if err != nil {
+		return nil, err
+	}
+	return encryption.New(sk, padding, 0, hashfunc), nil
+}
+
+func NewEncryptor(pub *ecdsa.PublicKey, salt []byte, padding int, hashfunc func() hash.Hash) (encryption.Encryptor, *ecdsa.PublicKey, error) {
+	privKey, err := crypto.GenerateSecp256k1Key()
```

Comment:
this generates an ephemeral key. it would be good to document this somewhere. it means that `NewEncrypter` returns a single-use encrypter if I understand correctly. I am not entirely familiar with el-gamal key exchange but if this is part of the protocol then it would still be nice to have this as part of the constructor documentation